File: Partition/Partition.java
import java.io.*;
import java.util.*;
import crio.ds.List.ListNode;

class Partition {

  public static void  main (String args []) {
    Scanner sc =  new  Scanner (System.in);

    int n = sc.nextInt();
    ArrayList <Integer> list = new ArrayList<Integer> ();
    int x;
    for ( int i = 0 ; i < n; i++) {
      x = sc.nextInt();
      list.add(x);
    }
    ListNode head = ListNode.createList(list);
    int k = sc.nextInt();
    if (n > 0)
      head = Solution.partition(head, k);

    List<Integer> nums = ListNode.extractList(head);

    for ( int i = 0; i < nums.size() ; i++) {
      System.out.print(nums.get(i)  + " ");
    }

  }

}

File: Partition/Partition.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"

int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
    }
    int x;
    cin >> x;

    ListNode* head = ListNode().createList(list);

    ListNode* result = partition(head , x);
    vector<int> resultList = ListNode().extractList(result);
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}
